service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/AddNoteController.java

```diff
@@ -26,21 +25,22 @@
 public class AddNoteController {
     private final CaseNoteService service;
     private final RequestData requestData;
-    private final ObjectMapper mapper;
+    private final CaseDataConverter caseDataConverter;
 
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
-        CaseDetails caseDetails = callbackRequest.getCaseDetails();
-        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+        CaseData caseData = caseDataConverter.convertToCaseData(callbackRequest);
 
         CaseNote caseNote = service.buildCaseNote(requestData.authorisation(), caseData.getCaseNote());
         List<Element<CaseNote>> caseNotes = service.addNoteToList(caseNote, caseData.getCaseNotes());
 
-        caseDetails.getData().put("caseNotes", caseNotes);
-        caseDetails.getData().remove("caseNote");
+        CaseData updatedCase = caseData.toBuilder()
+            .caseNotes(caseNotes)
+            .caseNote(null)
+            .build();
```

Comment:
this entire thing is over engineered tbh. 
1. No need to pass requestData.authorisation to service, service can inject RequestData on its own,
2. having addNoteToList which just add element to list is overengineered

Why not having just single method, public CaseData service.addNote(CaseDate caseData){
 //get note, enrich with user info, add to list, reset note, return case data
}

Reply:
as discussed, my aim of this Spike/PR is just to remove strings while building case data.